mobile/lib/src/infrastructure/connection_repositories/eeg_data_repository.dart

```diff
@@ -26,26 +29,23 @@ class EEGDataRepository implements IEEGDataRepository {
   BaseOpenBCITransformer<List<int>, List<dynamic>> currentStreamTransformer;
   BaseOpenBCITransformer<List<int>, List<dynamic>> _currentTransformer;
   StreamSubscription _currentTransformerStream;
-  StreamingSharedPreferences _preferences;
   EEGData _recordingData;
   double _fpzCzChannelMax = 0;
   double _pzOzChannelMax = 0;
   int _dataCount = 0;
 
   @override
-  void initialize() async {
-    _preferences ??= await StreamingSharedPreferences.instance;
-
-    _currentTransformer =
-        _preferences.getBool('using_bluetooth', defaultValue: false).getValue()
-            ? _ganglionTransformer
-            : _cytonTransformer;
+  void initialize(DeviceType deviceType) async {
+    _currentTransformer = (deviceType == DeviceType.bluetooth)
+        ? _ganglionTransformer
+        : _cytonTransformer;
   }
 
   @override
   void createRecordingFromStream(Stream<List<int>> stream) {
```

Comment:
I am wondering if we need EEGDataRepository at all. Maybe another repository's method could return something like a SleepSequenceRecordingController that has start and end method to get controll over stream. Remember that each and every repositories should be thightly linked to a single domaine entity.